### ISSUE TITLE
Pyrakoon not able to connect to cluster if one member is down

### ISSUE THREAD

**Issue by jeroenmaelbrancke** [openvstorage/pyrakoon#17]
# Problem
When one member of the Arakoon cluster is down the Pyrakoon is not able to find the master.

Error message of the pyrakoon client:
```
2018-06-25 23:52:39 64900 -0400 - SV4SRV0015 - 28624/139769160349440 - extensions/ovs_extensions.db.arakoon.pyrakoon.pyrakoon.compat - 2 - ERROR - No connection available to node at '172.26.16.7:26400' : Message exchange with node sDEjp3rMpJ2ACr4L failed
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/ovs_extensions/db/arakoon/pyrakoon/pyrakoon/compat.py", line 1160, in _send_message
    connection.send(data)
  File "/usr/lib/python2.7/dist-packages/ovs_extensions/db/arakoon/pyrakoon/pyrakoon/compat.py", line 1325, in send
    raise ArakoonNotConnected(self._address)
ArakoonNotConnected: No connection available to node at '172.26.16.7:26400'
2018-06-25 23:52:39 64900 -0400 - SV4SRV0015 - 28624/139769160349440 - extensions/ovs_extensions.db.arakoon.pyrakoon.pyrakoon.compat - 3 - ERROR - No connection available to node at '172.26.16.7:26400': Unable to query node "sDEjp3rMpJ2ACr4L" to look up master
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/ovs_extensions/db/arakoon/pyrakoon/pyrakoon/compat.py", line 1201, in determine_master
    self.master_id = self._get_master_id_from_node(node)
  File "/usr/lib/python2.7/dist-packages/ovs_extensions/db/arakoon/pyrakoon/pyrakoon/compat.py", line 1229, in _get_master_id_from_node
    connection = self._send_message(node_id, data)
  File "/usr/lib/python2.7/dist-packages/ovs_extensions/db/arakoon/pyrakoon/pyrakoon/compat.py", line 1160, in _send_message
    connection.send(data)
  File "/usr/lib/python2.7/dist-packages/ovs_extensions/db/arakoon/pyrakoon/pyrakoon/compat.py", line 1325, in send
    raise ArakoonNotConnected(self._address)
ArakoonNotConnected: No connection available to node at '172.26.16.7:26400'

Jun 25 23:52:45 SV4SRV0015 ovs-workers[28624]: 2018-06-25 23:52:45 79600 -0400 - SV4SRV0015 - 28624/139769160349440 - extensions/ovs_extensions.db.arakoon.pyrakoon.pyrakoon.compat - 6 - ERROR - timed out: Unable
 to connect to ('172.26.16.7', 26400)
Jun 25 23:52:45 SV4SRV0015 ovs-workers[28624]: Traceback (most recent call last):
Jun 25 23:52:45 SV4SRV0015 ovs-workers[28624]:   File "/usr/lib/python2.7/dist-packages/ovs_extensions/db/arakoon/pyrakoon/pyrakoon/compat.py", line 1284, in connect
Jun 25 23:52:45 SV4SRV0015 ovs-workers[28624]:     self._socket = socket.create_connection(self._address, self._timeout)
Jun 25 23:52:45 SV4SRV0015 ovs-workers[28624]:   File "/usr/lib/python2.7/socket.py", line 575, in create_connection
Jun 25 23:52:45 SV4SRV0015 ovs-workers[28624]:     raise err
Jun 25 23:52:45 SV4SRV0015 ovs-workers[28624]: timeout: timed out
```

While a master was found on the arakoon:
```
root@SV4SRV0001:~# arakoon-1.9.25 --who-master -config /opt/OpenvStorage/config/arakoon_config.ini
yBlo4YnVC2FOoWYy
```

**Comment by JeffreyDevloo** [comment]
Nodes in the ArakoonClientConfig:
```
{'Dvr5mMoOINJfz8Ac': (['172.26.16.4'], 26400),
 'TQKv2impGTvArKf7': (['172.26.16.5'], 26400),
 'kxeaWx7yLWCA46In': (['172.26.16.8'], 26400),
 'sDEjp3rMpJ2ACr4L': (['172.26.16.7'], 26400),
 'yBlo4YnVC2FOoWYy': (['172.26.16.6'], 26400)}
```

**Comment by toolslive** [comment]
```
import Compat
clusterId = 'ricky'
config = Compat.Arakoon.ArakoonClientConfig(
    clusterId,
    {
        "arakoon_0":("127.0.0.1",4000),
        "arakoon_1":("127.0.0.1",4001),
        "arakoon_2":("127.0.0.1",4002)})

client = Compat.Arakoon.ArakoonClient(config)
print client.whoMaster()
```

```
$> pgrep -a arakoon
# arakoon_0 was killed earlier 
5907 ./arakoon.native --node arakoon_1 -config ./cfg/arakoon.ini.mine
5908 ./arakoon.native --node arakoon_2 -config ./cfg/arakoon.ini.mine

$> python my_bug.py 
arakoon
WARNING:root:Unable to connect to 127.0.0.1:4000 (error: '[Errno 111] Connection refused')
WARNING:root:Unable to connect to 127.0.0.1:4000 (error: '[Errno 111] Connection refused')
WARNING:root:Attempt 0 to exchange message with node arakoon_0 failed with error (ArakoonNotConnected: 'No connection available to node at ['127.0.0.1'] on port 4000').
WARNING:root:Could not query node 'arakoon_0' to see who is master
arakoon_1
```

you get warnings,  but the master lookup succeeds.

**Comment by toolslive** [comment]
@wimpers : probably needs to be moved to `framework`, it points to something else.
